public/app/plugins/datasource/cloud-monitoring/module.ts

```diff
@@ -1,14 +1,65 @@
-import { DataSourcePlugin } from '@grafana/data';
+import { DataSourcePlugin, DashboardLoadedEvent } from '@grafana/data';
+import { getAppEvents } from '@grafana/runtime';
 
 import CloudMonitoringCheatSheet from './components/CloudMonitoringCheatSheet';
 import { ConfigEditor } from './components/ConfigEditor/ConfigEditor';
 import { QueryEditor } from './components/QueryEditor';
 import { CloudMonitoringVariableQueryEditor } from './components/VariableQueryEditor';
 import CloudMonitoringDatasource from './datasource';
-import { CloudMonitoringQuery } from './types';
+import pluginJson from './plugin.json';
+import { trackCloudMonitoringDashboardLoaded } from './tracking';
+import { CloudMonitoringQuery, QueryType } from './types';
 
 export const plugin = new DataSourcePlugin<CloudMonitoringDatasource, CloudMonitoringQuery>(CloudMonitoringDatasource)
   .setQueryEditorHelp(CloudMonitoringCheatSheet)
   .setQueryEditor(QueryEditor)
   .setConfigEditor(ConfigEditor)
   .setVariableQueryEditor(CloudMonitoringVariableQueryEditor);
+
+// Track dashboard loads to RudderStack
+getAppEvents().subscribe<DashboardLoadedEvent<CloudMonitoringQuery>>(
+  DashboardLoadedEvent,
+  ({ payload: { dashboardId, orgId, grafanaVersion, queries } }) => {
+    const cloudmonitorQueries = queries[pluginJson.id];
+    let stats = {
+      [QueryType.TIME_SERIES_QUERY]: {
+        hidden: 0,
+        visible: 0,
+      },
+      [QueryType.TIME_SERIES_LIST]: {
+        hidden: 0,
+        visible: 0,
+      },
+      [QueryType.SLO]: {
+        hidden: 0,
+        visible: 0,
+      },
+      [QueryType.ANNOTATION]: {
+        hidden: 0,
+        visible: 0,
+      },
+    };
+    cloudmonitorQueries.forEach((query) => {
+      if (
+        query.queryType === QueryType.TIME_SERIES_QUERY ||
+        query.queryType === QueryType.TIME_SERIES_LIST ||
+        query.queryType === QueryType.SLO ||
+        query.queryType === QueryType.ANNOTATION
```

Comment:
have you checked if this work with queries created prior to 9.4? For those queries, most of them would be the query type `metrics` and I don't know if by the point this function is executed, the queries would have been migrated or not (I suspect not).

If that's the case, you need to check if `editorMode` is `mql` or not to differentiate between the two query types (as done in the migration code).

Reply:
Great catch, I've updated to reflect non-migrated queries